### PR TITLE
Counting Estimator: Preliminary Task

### PR DESCRIPTION
**Pull request by SooryaN**
@unoebauer, here's the prelim task. Do get back to me on how close/far I am.

PS: The badass was a test to see if changes were being reflected. I do not believe it helps with the counting.

### CODE REVIEW

File: tardis/montecarlo/src/cmontecarlo.c [hunk 1]
```diff
@@ -683,6 +700,9 @@ montecarlo_line_scatter (rpacket_t * packet, storage_model_t * storage,
     {
       rpacket_set_tau_event (packet,
```
Comment (SooryaN) [inline tardis/montecarlo/src/cmontecarlo.c]: Here, am I right in assuming that tau refers to the photon depth?

**Comment by unoebauer** [comment]
Hi @SooryaN - thanks for the PR. 

The counter you implemented actually counts something different from what we want. If you look closely at what happens in the routine ``montecarlo_linescatter``, you'll see that your counter actually counts only packets which come into resonance but which **do not** interact.

Look again at the ideas page - I've added another hint for the objective. Digest the information there and then rethink the large block comment you introduced in the source code.

**Comment by SooryaN** [comment]
@unoebauer, I've made a couple of changes to the earlier one. I'd like to know your feedback.

File: tardis/montecarlo/src/cmontecarlo.c [hunk 2]
```diff
@@ -364,8 +372,35 @@ increment_j_blue_estimator (const rpacket_t * packet, storage_model_t * storage,
...
 #pragma omp atomic
 #endif
+
   storage->line_lists_j_blues[j_blue_idx] +=
-    comov_energy / rpacket_get_nu (packet);
+  comov_energy / rpacket_get_nu (packet);
+  double key = rpacket_get_nu (packet) * doppler_factor;
+  double *p2 = (double *) bsearch(&key,storage->line_list_nu , storage->no_of_lines, sizeof(storage->line_list_nu[0]), compare);
+  if (p2){
+      count_resonance++;
```
Comment (unoebauer) [inline tardis/montecarlo/src/cmontecarlo.c]: @SooryaN - the counter now resides at the most logical location in the program flow. However, you are making the counting task unnecessarily complicated. See comment to line 376!

File: tardis/montecarlo/src/cmontecarlo.c [hunk 3]
```diff
@@ -364,8 +372,35 @@ increment_j_blue_estimator (const rpacket_t * packet, storage_model_t * storage,
 #ifdef WITHOPENMP
 #pragma omp atomic
 #endif
+
   storage->line_lists_j_blues[j_blue_idx] +=
```
Comment (unoebauer) [inline tardis/montecarlo/src/cmontecarlo.c]: @SooryaN - try to figure out what this instruction does and why we are doing it. Adding a **line-specific** counter is then a matter of a few lines...

File: tardis/montecarlo/src/cmontecarlo.c [hunk 4]
```diff
@@ -7,8 +7,16 @@
 #include "abbrev.h"
 #include "cmontecarlo.h"
 #include "io.h"
-
-
+long long int count_resonance = 0;
+static int
+compare(const void *ap, const void *bp)
+{
```
Comment (unoebauer) [inline tardis/montecarlo/src/cmontecarlo.c]: @SooryaN - if you are using the current line interaction and resonance determination infrastructure of the code, you will never have to compare frequencies actually.

File: tardis/montecarlo/src/cmontecarlo.c [hunk 5]
```diff
@@ -7,8 +7,16 @@
 #include "abbrev.h"
 #include "cmontecarlo.h"
 #include "io.h"
-
-
+long long int count_resonance = 0;
```
Comment (unoebauer) [inline tardis/montecarlo/src/cmontecarlo.c]: @SooryaN - as a first step to completing the task, defining a global resonance counter is completely fine. However, keep in mind that eventually the goal is to count resonances on a per-line basis.

**Comment by SooryaN** [comment]
@unoebauer , Third time's the charm hopefully.

**Pull request closed without merging**